### PR TITLE
AWS Amplify DataStore

### PR DESCRIPTION
**Pull request by garrettluu**
Adding an adapter class that allows front-end to interact with AWS Amplify DataStore database

### CODE REVIEW

File: src/datastore/datastore.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        const old = (await DataStore.query(Room, room => room.id("eq", code)))[0];
+        await DataStore.save(
+            Room.copyOf(old, updated => {
+                updated.id = original.id,
+                    updated.restaurants = list,
+                    current = list[0]
+            })
+        );
+    }
+}
```
Comment (garrettluu) [inline src/datastore/datastore.js:60]: Are there any other types of queries we need?

File: .graphqlconfig.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    schemaPath: src/graphql/schema.json
+    includes:
+      - src/graphql/**/*.js
+    excludes:
+      - ./amplify/**
+    extensions:
+      amplify:
+        codeGenTarget: javascript
+        generatedFileName: ''
+        docsFilePath: src/graphql
```
Comment (aemmadi) [inline .graphqlconfig.yml:12]: Wait are we using GraphQL?

File: src/datastore/datastore.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    async createRoom(user) {
+        let id = Math.floor(Math.random() * (999999 - 100000 + 1)) + 100000;
+        await DataStore.save(new Room({
+            code: id,
+            restaurants: [],
+            users: [user],
+            current: null,
+        }));
+        return id;
+    }
```
Comment (aemmadi) [inline src/datastore/datastore.js:15]: I know the chances are slim, but in the case that we have 2 same id's generated randomly, do we purge the room once everyone leaves?

File: src/datastore/datastore.js (hunk 1 above)
Comment (aemmadi) [reply src/datastore/datastore.js:60]: I think you covered the main ones

File: .graphqlconfig.yml (hunk 2 above)
Comment (garrettluu) [reply .graphqlconfig.yml:12]: Yeah, AWS DataStore is based on GraphQL

File: src/datastore/datastore.js (hunk 3 above)
Comment (garrettluu) [reply src/datastore/datastore.js:15]: Good idea, I think I'll add a query that scans and removes all rooms with no users.

**Pull request merged**